src/overlays/actors/ovl_Bg_Po_Syokudai/z_bg_po_syokudai.c

```diff
@@ -21,11 +46,128 @@ const ActorInit Bg_Po_Syokudai_InitVars = {
+                    (((u8)this->unk_14C) << 8) + this->actor.params + 0x1000);
+
+    } else if (!Flags_GetSwitch(globalCtx, this->actor.params)) {
+        if (globalCtx->envCtx.unk_BF == 0xFF) {
+            globalCtx->envCtx.unk_BF = 0x4U;
+        }
+    }
+
+    this->unk_14E = (s16)(Math_Rand_ZeroOne() * 20.0f);
+}
+
+void BgPoSyokudai_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+
+    if (globalCtx->envCtx.unk_BF != 0xFF) {
+        globalCtx->envCtx.unk_BF = (u8)0xFFU;
+    }
+}
+
+void BgPoSyokudai_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    void* pad;
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        func_8002F974(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    this->unk_14E += 1;
+}
+
+void BgPoSyokudai_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    f32 lightBrightness;
+    u8 red;
+    u8 green;
+    u8 blue;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_po_syokudai.c", 315);
+
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_po_syokudai.c", 319),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(oGfxCtx->polyOpa.p++, D_060003A0);
+
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        Color_RGBA8* primColor = &primColors[this->unk_14C];
+        Color_RGBA8* envColor = &envColors[this->unk_14C];
+
+        lightBrightness = (0.3f * Math_Rand_ZeroOne()) + 0.7f;
+
+        red = ((u32)(f32)(primColor->r * lightBrightness)) & 0xFF;
+        green = ((u32)(f32)(primColor->g * lightBrightness)) & 0xFF;
+        blue = ((u32)((f32)primColor->b * lightBrightness)) & 0xFF;
+
+        Lights_PointSetColorAndRadius(&this->lightInfo, red, green, blue, 0xC8);
+
+        func_80093D84(globalCtx->state.gfxCtx);
+        gSPSegment(oGfxCtx->polyXlu.p++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0U, 0U, 0x20, 0x40, 1, 0,
+                                    ((this->unk_14E) * -20) & 0x1FF, 0x20, 0x80));
+
+        gDPSetPrimColor(oGfxCtx->polyXlu.p++, 0x80, 0x80, primColor->r, primColor->g, primColor->b, 0xFF);
+        gDPSetEnvColor(oGfxCtx->polyXlu.p++, envColor->r, envColor->g, envColor->b, 0xFF);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Syokudai/BgPoSyokudai_Update.s")
+        Matrix_Translate(0.0f, 52.0f, 0.0f, 1U);
+        Matrix_RotateY(
+            (s16)(func_8005A9F4(globalCtx->cameraPtrs[globalCtx->activeCamera]) - this->actor.shape.rot.y + 0x8000) *
+                0.0000958738f,
```

Comment:
`0.0000958738` is `(M_PI / 0x8000)` (`M_PI` being a constant for pi defined somewhere through the headers that are already included, no need to include anything)
for matching you may need to use `(M_PI / (f32)0x8000)` or `(M_PI / 32768.0f)`

( https://www.wolframalpha.com/input/?i=0.0000958738 )